.github/workflows/doxygen.yml

```diff
@@ -0,0 +1,30 @@
+name: doxygen
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-and-deploy:
+    name: Build and deploy HTML
+    runs-on: ubuntu-latest
+    steps:
+    - name: Fetching sources
+      uses: actions/checkout@v2
+    - name: Installing build dependencies
+      run: |
+        sudo apt update
+        sudo apt install cmake doxygen gcc git
+    - name: Creating build directory
+      run: |
+        mkdir build
+    - name: Building HTML using Doxygen
+      run: |
+        cmake -S . -B build -DBENCHMARK_ENABLE_TESTING:BOOL=OFF -DBENCHMARK_ENABLE_DOXYGEN:BOOL=ON
+        cmake --build build --target benchmark_doxygen
+    - name: Deploying built HTML to GitHub pages
+      uses: JamesIves/github-pages-deploy-action@4.0.0
+      with:
+        branch: gh-pages
```

Comment:
our pages are linked to main, iiuc, and there is no gh-pages branch.

also this would override the docs that are already in `docs/` i think.. wouldn't it?

Reply:
I think that pushing a lot of dynamically generated static assets to `docs` directory is not a good idea.

Reply:
but are the results of this push accessible anywhere if our project is set up with `main` as the pages branch?

Reply:
I have a better idea. I will push changes to my fork shortly and we can see the final results.

Reply:
Site index: https://xvitaly.github.io/benchmark/

Static page example: https://xvitaly.github.io/benchmark/md_docs_user_guide.html

Reply:
some nitpicks: the main page doesn't have working links, and the user guide code samples aren't syntax formatted. i don't know if these are solvable.

i think what i would expect as an end state is the existing docs working as they are (https://google.github.io/benchmark) with a separate location (https://google.github.io/benchmark/api or /dev ?) that has just the generated code documentation.

or, we can just have the simple version where there's an option to generate doxygen locally for people that want it and leave it there.

Reply:
> some nitpicks: the main page doesn't have working links, and the user guide code samples aren't syntax formatted. i don't know if these are solvable.

Very old doxygen version, I think. Doxygen 1.9.1 from Fedora 34 generate working links.

> i think what i would expect as an end state is the existing docs working as they are

All these docs already built with Doxygen and pushed to the website.

> or, we can just have the simple version where there's an option to generate doxygen locally for people that want it and leave it there.

I like this option.

Reply:
> > or, we can just have the simple version where there's an option to generate doxygen locally for people that want it and leave it there.
> 
> I like this option.

Sure, that sounds good.

Reply:
> Sure, that sounds good.

Removed GitHub actions file and disabled the Doxygen target by default.

Reply:
Also we can install Doxygen files to `${CMAKE_CURRENT_BINARY_DIR}` and disable installation of files from `docs/` directory, because generated by Doxygen HTMLs already includes all of them.